src/index.ts

```diff
@@ -73,37 +92,33 @@ export const compare = memoize((versionA: VersionInput, versionB: VersionInput):
 	versionA = normalize(versionA);
 	versionB = normalize(versionB);
 
-	const isAValid = semver.valid(versionA);
-	const isBValid = semver.valid(versionB);
-	if (isAValid && !isBValid) {
-		return 1;
-	}
-	if (!isAValid && isBValid) {
-		return -1;
-	}
-	if (!isAValid && !isBValid) {
-		if (versionA > versionB) {
+	const semverA = semver.parse(versionA);
+	const semverB = semver.parse(versionB);
+	if (!semverA || !semverB) {
+		if (semverA) { // !semverB
 			return 1;
 		}
-		if (versionA < versionB) {
+		if (semverB) { // !semverA
 			return -1;
 		}
-		return 0;
+		return compareValues(versionA, versionB);
 	}
-	const semverResult = semver.compare(versionA, versionB);
+
+	const semverResult = semver.compare(semverA, semverB);
 	if (semverResult !== 0) {
 		return semverResult;
 	}
-	const revA = getRev(versionA);
-	const revB = getRev(versionB);
-	if (revA !== revB) {
-		return revA > revB ? 1 : -1;
+
+	const revResult = compareValues(getRev(semverA), getRev(semverB));
+	if (revResult !== 0) {
+		return revResult;
 	}
-	const devA = Number(isDevelopmentVersion(versionA));
-	const devB = Number(isDevelopmentVersion(versionB));
-	if (devA !== devB) {
-		return devA > devB ? -1 : 1;
+
+	const devResult = -compareValues(isDevelopmentVersion(semverA), isDevelopmentVersion(semverB));
```

Comment:
Nit: the `-` is a bit subtle here. Could we make that more explicit and put it below? e.g. `return devResult * -1`